cmd/kubeadm/app/cmd/phases/upgrade/node/controlplane.go

```diff
@@ -78,6 +79,12 @@ func runControlPlane() func(c workflow.RunData) error {
 			return errors.Wrap(err, "couldn't complete the static pod upgrade")
 		}
 
+		if features.Enabled(cfg.FeatureGates, features.UpgradeAddonsAfterControlPlane) {
```

Comment:
We cannot remove this check now. Because the check in `PerformAddonsUpgrade` only affects whether to wait for all controlplane nodes to be upgraded, when this featuregate is disabled or the check passed, `PerformAddonsUpgrade` will upgrades addons immediately. But when the featuregate is disabled, we don't want to upgrade addons in `upgrade node`, because we already did that with `upgrade apply`.